src/log-store/src/kafka/client_manager.rs

```diff
@@ -134,3 +134,77 @@ impl ClientManager {
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use common_test_util::get_broker_endpoints;
+    use common_test_util::wal::kafka::topic_decorator::{Affix, TopicDecorator};
+    use common_test_util::wal::kafka::{create_topics, BROKER_ENDPOINTS_KEY};
+
+    use super::*;
+
+    /// Prepares for a test in that a collection of topics and a client manager are created.
+    async fn prepare(test_name: &str, num_topics: usize) -> (ClientManager, Vec<Topic>) {
+        let broker_endpoints = get_broker_endpoints!(BROKER_ENDPOINTS_KEY);
+        let decorator = TopicDecorator::default()
+            .with_prefix(Affix::Fixed(test_name.to_string()))
+            .with_suffix(Affix::TimeNow);
+        let topics = create_topics(num_topics, decorator, &broker_endpoints, None).await;
+
+        let config = KafkaConfig {
+            broker_endpoints,
+            ..Default::default()
+        };
+        let manager = ClientManager::try_new(&config).await.unwrap();
+
+        (manager, topics)
+    }
+
+    /// Checks clients for the given topics are created.
+    async fn ensure_clients_exist(topics: &[Topic], client_manager: &ClientManager) {
+        let client_pool = client_manager.client_pool.read().await;
+        let all_exist = topics.iter().all(|topic| client_pool.contains_key(topic));
+        assert!(all_exist);
+    }
+
+    async fn test_with(test_name: &str) {
+        let (manager, topics) = prepare(test_name, 128).await;
+        // Assigns multiple regions to a topic.
+        let region_topic = (0..512)
+            .map(|region_id| (region_id, &topics[region_id % topics.len()]))
+            .collect::<HashMap<_, _>>();
+
+        // Gets the assigned topic for each region and then gets the associated client.
+        match test_name {
+            "test_sequential" => {
+                // Gets all clients sequentially.
+                for (_, topic) in region_topic {
+                    manager.get_or_insert(topic).await.unwrap();
+                }
+            }
+            "test_parallel" => {
+                // Gets all clients in parallel.
+                let tasks = region_topic
+                    .values()
+                    .map(|topic| manager.get_or_insert(topic))
+                    .collect::<Vec<_>>();
+                futures::future::try_join_all(tasks).await.unwrap();
+            }
+            _ => unreachable!(),
+        }
+        // Ensures all clients are created successfully.
+        ensure_clients_exist(&topics, &manager).await;
+    }
+
+    /// Sends `get_or_insert` requests sequentially to the client manager, and checks if it could handle them correctly.
+    #[tokio::test]
+    async fn test_sequential() {
+        test_with("test_sequential").await;
+    }
+
+    /// Sends `get_or_insert` requests in parallel to the client manager, and checks if it could handle them correctly.
+    #[tokio::test]
+    async fn test_parallel() {
+        test_with("test_parallel").await;
+    }
```

Comment:
Let's expense the tests. 
For reason: 
- Tests must be easy to read and modify; it's by design to have duplicated code.
- Every test case should be separate so we can understand each other without reading another part of the code. (Of course, we can add some utils for complex assertions, common setups, etc.)

```suggestion
    #[tokio::test]
    async fn test_sequential() {
        let (manager, topics) = prepare(test_name, 128).await;
        // Assigns multiple regions to a topic.
        let region_topic = (0..512)
            .map(|region_id| (region_id, &topics[region_id % topics.len()]))
            .collect::<HashMap<_, _>>();

        // Gets all clients sequentially.
        for (_, topic) in region_topic {
            manager.get_or_insert(topic).await.unwrap();
        }
        // Ensures all clients are created successfully.
        ensure_clients_exist(&topics, &manager).await;
    }

    /// Sends `get_or_insert` requests in parallel to the client manager, and checks if it could handle them correctly.
    #[tokio::test(flavor = "multi_thread")]
    async fn test_parallel() {
        let (manager, topics) = prepare("test_parallel", 128).await;
        let manager = Arc::new(manager);
        // Assigns multiple regions to a topic.
        let region_topic = (0..512)
            .map(|region_id| (region_id, &topics[region_id % topics.len()]))
            .collect::<HashMap<_, _>>();

        // Gets the assigned topic for each region and then gets the associated client.
        // Gets all clients in parallel.
        let handlers = region_topic
            .values()
            .map(|topic| {
                let manager_moved = manager.clone();
                let topic = topic.to_string();
                tokio::spawn(async move { manager_moved.get_or_insert(&topic).await })
            })
            .collect::<Vec<_>>();
        let _ = futures::future::try_join_all(handlers)
            .await
            .unwrap()
            .into_iter()
            .collect::<Result<Vec<_>>>()
            .unwrap();

        // Ensures all clients are created successfully.
        ensure_clients_exist(&topics, &manager).await;
    }
```